syft/workers/base.py

```diff
@@ -669,15 +688,22 @@ def de_register_obj(self, obj: object, _recurse_torch_objs: bool = True):
 
     # SECTION: convenience methods for constructing frequently used messages
 
-    def send_obj(self, obj: object, location: "BaseWorker"):
+    def send_obj(self, obj: object, location: "BaseWorker", requires_grad=False):
```

Comment:
Wonder if we could avoid the binary flag if the `sender` and `origin_id` were already set on the `object` by the time this method got called. 🤔

Reply:
I handle the requires_grad kwarg at the worker.send() level now